internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -81,12 +81,15 @@ func (q amsQuotaService) newBaseQuotaReservedResourceBuilder(kafka *dbapi.KafkaR
 		rr.ResourceType(resourceType)
 	}
 	rr.ResourceName(ocm.RHOSAKResourceName)
-	rr.BillingModel(amsv1.BillingModelMarketplace)
 	rr.Count(1)
 	return *rr
 }
 
-var supportedAMSBillingModels map[string]struct{} = map[string]struct{}{
+// supportedAMSRelatedResourceBillingModels returns the supported AMS billing
+// models for AMS RelatedResources. An AMS RelatedResource is not to be confused
+// with an AMS ReservedResource. The set of Billing models shown/accepted is
+// different between them
+var supportedAMSRelatedResourceBillingModels map[string]struct{} = map[string]struct{}{
```

Comment:
double type declaration, can be shortened to `var supportedAMSRelatedResourceBillingModels = map[string]struct{}{`

Reply:
Changed